server/src/services/domain/user.service.ts

```diff
@@ -0,0 +1,88 @@
+import * as i18n from 'i18n';
+import { StorageService } from './storage.service';
+import { UserStorage } from '../../models/storage.models';
+import { User } from '../../models/user.model';
+import { catchAsyncError } from '../../utils/catchError';
+import { logger } from '../../utils/logger';
+import { LogCategory } from '../../models/constants';
+
+export class UserService {
+    private users: UserStorage;
+
+    constructor(private storage: StorageService) {
+    }
+
+    public getUser(user: number): Promise<User | null>;
+    public getUser(user: User): Promise<User | null>;
+    public getUser(user: User | number): Promise<User | null> {
+        return Promise.resolve(
+            this.users[
+                Object.keys(this.users).find(
+                    (key) => typeof user === 'number'
+                        ? parseInt(key, 10) === user
+                        : this.users[key].chatId === user.chatId,
+                )
+                ],
+        );
+    }
+
+    public async setUserLocale(
+        user: User,
+        locale: string,
+    ): Promise<void> {
+        return this.storage.addUser({
+            ...user,
+            settings: {
+                ...user.settings,
+                locale,
+            },
+        });
+    }
+
+    public async setUserInterruptedCommand(
```

Comment:
@danbilokha what is interrupted command?

Reply:
discussed